unittests/whitelist_blacklist_tests.cpp

```diff
@@ -25,42 +25,17 @@ class whitelist_blacklist_tester {
    public:
       whitelist_blacklist_tester() {}
 
-      static controller::config get_default_chain_configuration( const fc::path& p ) {
-         controller::config cfg;
-         cfg.blocks_dir = p / config::default_blocks_dir_name;
-         cfg.state_dir  = p / config::default_state_dir_name;
-         cfg.state_size = 1024*1024*8;
-         cfg.state_guard_size = 0;
-         cfg.reversible_cache_size = 1024*1024*8;
-         cfg.reversible_guard_size = 0;
-         cfg.contracts_console = true;
-
-         cfg.genesis.emplace();
-         cfg.genesis->initial_timestamp = fc::time_point::from_iso_string("2020-01-01T00:00:00.000");
-         cfg.genesis->initial_key = base_tester::get_public_key( config::system_account_name, "active" );
-
-         for(int i = 0; i < boost::unit_test::framework::master_test_suite().argc; ++i) {
-            if(boost::unit_test::framework::master_test_suite().argv[i] == std::string("--wavm"))
-               cfg.wasm_runtime = chain::wasm_interface::vm_type::wavm;
-            else if(boost::unit_test::framework::master_test_suite().argv[i] == std::string("--wabt"))
-               cfg.wasm_runtime = chain::wasm_interface::vm_type::wabt;
-         }
-
-         return cfg;
-      }
-
       void init( bool bootstrap = true ) {
          FC_ASSERT( !chain, "chain is already up" );
 
-         auto cfg = get_default_chain_configuration( tempdir.path() );
-         cfg.sender_bypass_whiteblacklist = sender_bypass_whiteblacklist;
-         cfg.actor_whitelist = actor_whitelist;
-         cfg.actor_blacklist = actor_blacklist;
-         cfg.contract_whitelist = contract_whitelist;
-         cfg.contract_blacklist = contract_blacklist;
-         cfg.action_blacklist = action_blacklist;
-
-         chain.emplace(cfg);
+         chain.emplace(tempdir, [&](controller::config& cfg) {
+            cfg.sender_bypass_whiteblacklist = sender_bypass_whiteblacklist;
+            cfg.actor_whitelist = actor_whitelist;
+            cfg.actor_blacklist = actor_blacklist;
+            cfg.contract_whitelist = contract_whitelist;
+            cfg.contract_blacklist = contract_blacklist;
+            cfg.action_blacklist = action_blacklist;
```

Comment:
Once again we have a similar issue as in `api_tests/more_deferred_transaction_tests` where currently the base_tester has `contracts_console == true` but the `validating_node` does not, and in this new behavior:

- If `Tester` is `tester`, then everything would still be the same assuming you make the changes to the `default_config` I specified in  `tester.hpp`.
- If `Tester` is `validating_tester` then both nodes will have `contracts_console == false`.

One option to ensure that console data is printed (for debugging purposes) is to set `cfg.contracts_console = true;` in this lambda here. But again, that would be slightly different behavior than before because then both nodes will have `contracts_console == true` whereas before the `validating_node` did not.

Reply:
Ultimately, I don't think it matters.  You seem to assume we got to this point because of planning, I think there was very little.  I think setting it to true for the main tester chain and false for the validation chain is best, next would be setting both to true (will not matter for general use, because we don't see the output even on failure, only when you run unittest -- --verbose), and trying to make it like it was before I started would just add to confusion with no benefit.